Notifier/composer.json

```diff
@@ -0,0 +1,42 @@
+{
+    "name": "msp/module-notifier",
+    "description": "Multiple channel notifier engine for Magento 2",
+    "require": {
+        "php": "~7.2.0 || ~7.3.0",
+        "magento/magento2-base": "^2.3",
```

Comment:
This constraint is also not correct:
1. `^2.3` would match unreleased major Magento versions, we can't be sure if this module will be fully compatible yet with those versions.
2. It's better to define the actual Magento modules which this Notifier module relies on. Since those modules follow semver versions, it's much safer to use those. Also: since Magento is moving more and more in the direction of making modules independent where in the future these modules would get updatable without the Magento marketing version to be increased, it makes sense to only depend on specific modules.

(same remark counts for the other composer.json files in this PR)

Reply:
That is correct. Just changed to `^2.3.0`

Reply:
Thanks, but still not correct, because `^2.3` and `^2.3.0` are 100% the same.

It's best to go for individual Magento modules and not target the `magento2-base` module.
See this example: https://github.com/magento/magento2/blob/2.3.2/app/code/Magento/CatalogUrlRewrite/composer.json#L9-L17

Reply:
Isn't the same in this case?

Reply:
Oh, you are right, I was messing with the `0` version behavior.
Now I see what you mean, sorry... you mean to use `2.3.*`, is that correct?

Reply:
Right, but it's not a good idea to depend on the "marketing" version of Magento itself, so I would remove that constraint on the `magento2-base` package entirely. Just add the actual magento modules in the requirements which this module is dependent on (for example: `magento/framework`, `magento/module-backend`, `magento/module-ui`, ...)

Reply:
Good point, thank you again for your valuable support.

Reply:
Now it should be correct.